qadence/ml_tools/train_grad.py

```diff
@@ -217,8 +240,19 @@ def loss_fn(model: torch.nn.Module, data: torch.Tensor) -> tuple[torch.Tensor, d
             except KeyboardInterrupt:
                 logger.info("Terminating training gracefully after the current iteration.")
                 break
+        try:
+            xs = next(dl_iter) if dataloader is not None else None  # type: ignore[arg-type]
+            xs_to_device = data_to_device(xs, device=device, dtype=data_dtype)
+            loss, metrics = loss_fn(model, xs_to_device)
+            if "val_loss" in metrics:
+                del metrics["val_loss"]
+            if iteration % config.print_every == 0 and config.verbose:
+                print_metrics(loss, metrics, iteration)
+
+        except KeyboardInterrupt:
+            logger.info("Terminating training gracefully after the current iteration.")
 
-    # Final writing and checkpointing
+    # Final printing, writing and checkpointing
     if config.folder and not config.checkpoint_best_only:
         write_checkpoint(config.folder, model, optimizer, iteration)
     write_tensorboard(writer, loss, metrics, iteration)
```

Comment:
Currently, the final loss is logged by default regardless of `config.write_every`. Should we consider logging the final validation loss by default as well?